lib/_stream_wrap.js

```diff
@@ -47,12 +44,12 @@ function StreamWrap(stream) {
     self.emit('error', err);
   });
   this.stream.on('data', function ondata(chunk) {
-    if (!(chunk instanceof Buffer)) {
+    if (typeof chunk === 'string' || this._readableState.objectMode === true) {
```

Comment:
The object mode check could actually be moved in the `StreamWrap` constructor but it is currently possible to change the `objectMode` variable from an existing stream and in that case the upper check would not be sufficient anymore.
It is explicitly noted in the docs that this is not safe but I'm thinking about opening a PR to make the `objectMode` read only.

@mcollina do you think that's reasonable?